File: web/di_core/DataIngestion/Domain/Response/GA4Source.ts
import { DataSources } from '@core/DataIngestion';
import { DataSource } from '@core/DataIngestion/Domain/Response/DataSource';
import { SourceId } from '@core/domain';

export class GA4Source implements DataSource {
  readonly className = DataSources.GA4Source;
  id: SourceId;
  orgId: string;
  displayName: string;
  refreshToken: string;
  accessToken: string;
  lastModify: number;

  constructor(id: SourceId, orgId: string, displayName: string, refreshToken: string, accessToken: string, lastModify: number) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.refreshToken = refreshToken;
    this.accessToken = accessToken;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): GA4Source {
    return new GA4Source(obj.id, obj.orgId, obj.displayName, obj.refreshToken, obj.accessToken, obj.lastModify);
  }
}

File: job-scheduler/src/main/scala/datainsider/jobscheduler/domain/source/Ga4Source.scala
package datainsider.jobscheduler.domain.source

import datainsider.jobscheduler.domain.{DataSource, DataSourceType}
import datainsider.jobscheduler.domain.DataSourceType.DataSourceType
import datainsider.jobscheduler.domain.Ids.SourceId

case class Ga4Source(
    orgId: Long,
    id: SourceId,
    displayName: String,
    refreshToken: String,
    accessToken: String,
    creatorId: String = "",
    lastModify: Long = System.currentTimeMillis()
) extends DataSource {
  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.Ga4

  override def getConfig: Map[String, Any] =
    Map(
      "access_token" -> accessToken,
      "refresh_token" -> refreshToken,
    )

  override def getCreatorId: String = creatorId

  override def getLastModify: SourceId = lastModify
}
